rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionEnabled.java

```diff
@@ -0,0 +1,82 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.rpc.modules.eth;
+
+import co.rsk.config.RskSystemProperties;
+import co.rsk.core.RskAddress;
+import co.rsk.core.Wallet;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.core.Account;
+import org.ethereum.core.Transaction;
+import org.ethereum.core.TransactionPool;
+import org.ethereum.facade.Ethereum;
+import org.ethereum.rpc.TypeConverter;
+import org.ethereum.rpc.Web3;
+import org.ethereum.vm.GasCost;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigInteger;
+
+import static org.ethereum.rpc.TypeConverter.stringHexToByteArray;
+
+public class EthModuleTransactionEnabled implements EthModuleTransaction {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("web3");
+
+    private final RskSystemProperties config;
+    private final Ethereum eth;
+    private final Wallet wallet;
+    private final TransactionPool transactionPool;
+
+    public EthModuleTransactionEnabled(RskSystemProperties config, Ethereum eth, Wallet wallet, TransactionPool transactionPool) {
+        this.config = config;
+        this.eth = eth;
+        this.wallet = wallet;
+        this.transactionPool = transactionPool;
+    }
+
+    @Override
+    public synchronized String sendTransaction(Web3.CallArguments args) {
+        Account account = this.wallet.getAccount(new RskAddress(args.from));
+        String s = null;
+        try {
+            String toAddress = args.to != null ? Hex.toHexString(stringHexToByteArray(args.to)) : null;
+
+            BigInteger value = args.value != null ? TypeConverter.stringNumberAsBigInt(args.value) : BigInteger.ZERO;
+            BigInteger gasPrice = args.gasPrice != null ? TypeConverter.stringNumberAsBigInt(args.gasPrice) : BigInteger.ZERO;
+            BigInteger gasLimit = args.gas != null ? TypeConverter.stringNumberAsBigInt(args.gas) : BigInteger.valueOf(GasCost.TRANSACTION_DEFAULT);
+
+            if (args.data != null && args.data.startsWith("0x")) {
+                args.data = args.data.substring(2);
+            }
+
+            synchronized (transactionPool) {
+                BigInteger accountNonce = args.nonce != null ? TypeConverter.stringNumberAsBigInt(args.nonce) : transactionPool.getPendingState().getNonce(account.getAddress());
+                Transaction tx = Transaction.create(config, toAddress, value, accountNonce, gasPrice, gasLimit, args.data);
+                tx.sign(account.getEcKey().getPrivKeyBytes());
+                eth.submitTransaction(tx.toImmutableTransaction());
+                s = tx.getHash().toJsonString();
+            }
+            return s;
+        } finally {
+            LOGGER.debug("eth_sendTransaction({}): {}", args, s);
```

Comment:
What about moving this line as the first line of the method and then removing the whole `try/finally` block?

Reply:
but then he wouldn't be able to log `s`. this is a pattern we inherited from EthereumJ that I don't like either, because it's not easy to see, but changing it is not that simple.